### DISCUSSION TITLE
TypeScript questions regarding type augmentation in NX project and problems with added session properties

### DISCUSSION THREAD

**Discussion by ColonelCordier** [nextauthjs/next-auth#6915] [Help]
### Question 💬

Hi,

We've a large project using NX, NextJS, Next-Auth and Auth0 as a provider.
I updated yesterday NextAuth (@latest) since the `unstable_getServerSideProps` has been stabilized. Since then, the project doesn't build because of TS errors in the callbacks functions when I set some properties in the session object.

![image](https://user-images.githubusercontent.com/103988855/223670684-ff988c6a-81c0-4f39-9207-0742324dd8ed.png)

Also, we were using the `email_verified` property from Auth0 (in `profile` object) to set an unverified `Role` in our app. This is not working with the update anymore (but the property from Auth0 still exists since I can log the `profile` object when I sign in).

![image](https://user-images.githubusercontent.com/103988855/223671992-8d8c07d3-17e7-4755-874b-e6c91d846be7.png)

Another question is about module augmentation, reading here https://next-auth.js.org/getting-started/typescript#module-augmentation, but I can't make it work in our NX project, that is architectured like this :

```
- apps
  - OUR_APPLICATION (NextJS)
- libs
  - feature_authentification
    - next-auth-config.ts
```
 
Where should I put the `types/next-auth.d.ts` file ?

Thank you
  

### How to reproduce ☕️

-

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by balazsorban44** [comment]
It doesn't matter where you augment a module, as long as the file is included in `tsconfig.json#includes`

Hard to tell more as there was no reproduction given.

**Comment by pencilcheck** [comment]
i have the same problem, added types/next-auth.d.ts, and it just doesn't pick up, keep complaining about the session.user doesn't have the attribute I specified.


----


seems like if I put the type augmentation directly next to the auth options variable file it worked.

looks like .d.ts just never get picked up when it clearly stated with wildcard in include that it should pick up.

**Reply by azeek21** [reply]
have you tried to set `typeRoots` in  your `tsconfig.json`. this did the trick for me.
